api/envoy/api/v2/listener/listener.proto

```diff
@@ -57,7 +57,7 @@ message Filter {
 // [#comment:TODO(PiotrSikora): destination IP / ranges are going to be 1.]
 // 1. Server name (e.g. SNI for TLS protocol),
 // 2. Transport protocol.
-// [#comment:TODO(PiotrSikora): application protocols are going to be 4.]
+// 3. Application protocols (e.g. ALPN for TLS protocol).
```

Comment:
There was some discussion yesterday about field precedence in matching. Should we resolve that before adding more fields? It will be harder to change the longer we wait. @htuch @mattklein123 .

Reply:
I don't think we have a consensus for the change yet, do we?

Also, if we're going to make a change, then I don't think it matters whether it's done for `transport_protocol` and `sni_domains` only or also for `application_protocol`, so I don't see a reason to block this PR on the new ordering rules.

Reply:
We might as well resolve the discussion in the other issue re: how we want to do matching. I think we can get resolution by Monday or Tuesday of next week (and can discuss in community meeting on Tuesday if needed).